packages/yarnpkg-shell/sources/index.ts

```diff
@@ -124,7 +124,10 @@ const BUILTINS = new Map<string, ShellBuiltin>([
     if (!stat.isDirectory())
       throw new ShellError(`cd: not a directory: ${target}`);
 
+    state.environment.PWD = resolvedTarget;
+    state.environment.OLDPWD = state.cwd;
```

Comment:
Don't they need to go through `npath.fromPortablePath` for portability, since they're exposed to third-party tools?